packages/core/src/sync-store/postgres/store.ts

```diff
@@ -161,24 +164,32 @@ export class PostgresSyncStore implements SyncStore {
       const fragments = buildLogFilterFragments({ ...logFilter, chainId });
 
       // First, attempt to merge overlapping and adjacent intervals.
-      await Promise.all(
-        fragments.map(async (fragment) => {
-          return await this.db.transaction().execute(async (tx) => {
+      for (const fragment of fragments) {
+        await this.db.transaction().execute(async (tx) => {
+          while (true) {
             const { id: logFilterId } = await tx
               .insertInto("logFilters")
               .values(fragment)
               .onConflict((oc) => oc.column("id").doUpdateSet(fragment))
               .returningAll()
               .executeTakeFirstOrThrow();
 
-            const existingIntervalRows = await tx
+            const existingIntervals = await tx
```

Comment:
Might be worth adding a comment here, "This is a trick to add a LIMIT to a DELETE statement"